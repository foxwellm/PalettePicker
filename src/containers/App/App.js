import React, { Component } from 'react';
import { Header } from '../Header/Header';
import { Palette } from '../Palette/Palette';
import { ControlForm } from '../ControlForm/ControlForm';
import { ProjectContainer } from '../ProjectContainer/ProjectContainer';
import '../../main.scss'

export class App extends Component {
  render() {
    return (
      <div>
        <Header />
        <Palette />
        <ControlForm />
        <ProjectContainer />
      </div>
    );
  }
}

export default App;
