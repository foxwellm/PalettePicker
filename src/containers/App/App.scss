*,
*:after,
*:before {
  box-sizing: border-box;
}

body {
  margin: 0;
  padding: 0;
  font-family: 'Open Sans', sans-serif;
}

.standard-button {
  padding: 8px;
  font-size: 14px;
  font-family: 'Open Sans', sans-serif;
  background-color: lightblue;
  border: 1px solid black;
  border-radius: 10px;
  box-shadow: 2px 2px 8px grey;

  &:hover:enabled {
    background-color: pink;
    box-shadow: none;
  }
}

input {
  padding: 8px;
  font-size: 14px;
  margin: 0 10px;
  width: 200px;
}

select {
  padding: 8px;
  font-size: 14px;
  width: 142px;
  background-color: white;
}

img {
  width: 25px;
}

button,
input,
select {
  &:focus {
    outline: none;
  }
}