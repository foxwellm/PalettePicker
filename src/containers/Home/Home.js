import React, { Component } from 'react';
import { connect } from 'react-redux';
import Palette from '../Palette/Palette';
import ControlForm from '../ControlForm/ControlForm';
import ProjectContainer from '../ProjectContainer/ProjectContainer';
import { fetchProjects } from '../../thunks/fetchProjects';
import PropTypes from 'prop-types';

export class Home extends Component {
  
  componentDidMount = () => {
    this.props.fetchProjects();
  };

  render() {
    const { searching } = this.props;
    return (
      <div>
        {!searching && <Palette />}
        {!searching && <ControlForm />}
        <ProjectContainer />
      </div>
    );
  };
};

export const mapStateToProps = (state) => ({
  projects: state.projects,
  searching: state.searching,
});

export const mapDispatchToProps = (dispatch) => ({
  fetchProjects: () => dispatch(fetchProjects()),
});

Home.propTypes = {
  projects: PropTypes.array,
  searching: PropTypes.bool,
  fetchProjects: PropTypes.func,
}

Home.defaultProps = {
  projects: [],
  searching: false,
}

export default connect(mapStateToProps, mapDispatchToProps)(Home);