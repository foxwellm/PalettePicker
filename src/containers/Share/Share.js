import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchProject } from '../../thunks/fetchProject';
import { fetchPalette } from '../../thunks/fetchPalette';
import ProjectContainer from '../ProjectContainer/ProjectContainer';
import Palette from '../Palette/Palette';
import NotFound from '../../components/NotFound/NotFound';
import PropTypes from 'prop-types';

export class Share extends Component {

  componentDidMount = () => {
    const { id, isProject, fetchProject, fetchPalette } = this.props;
    isProject ? fetchProject(id) : fetchPalette(id);
  };

  render() {
    const { projects, palettes, isProject, loading } = this.props;
    const palette = palettes[0];
    if(loading) {
      return <div>loading</div>
    } else if (isProject && projects.length) {
      return (
        <div>
          <ProjectContainer />
        </div>
      );
    } else if (!isProject && palettes.length) {
      return <Palette palette={palette} />
    } else {
      return <NotFound />
    }
  };
};

export const mapStateToProps = (state) => ({
  projects: state.projects,
  palettes: state.palettes,
  loading: state.loading,
});

export const mapDispatchToProps = (dispatch) => ({
  fetchProject: (id) => dispatch(fetchProject(id)),
  fetchPalette: (id) => dispatch(fetchPalette(id)),
});

Share.propTypes = {
  projects: PropTypes.array,
  palettes: PropTypes.array,
  fetchProject: PropTypes.func,
  fetchPalette: PropTypes.func,
}

Share.defaultProps = {
  palettes: [],
  projects: [],
}

export default connect(mapStateToProps, mapDispatchToProps)(Share);

