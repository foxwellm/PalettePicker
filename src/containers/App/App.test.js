import React from 'react';
import { App } from './App';
import { Home } from '../Home/Home';
import { Share } from '../Share/Share';
import { NotFound } from '../../components/NotFound/NotFound';
import { shallow, mount } from 'enzyme'
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import rootReducer from '../../reducers/index';
import thunk from 'redux-thunk';

describe('App', () => {
  const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(thunk)));
  let wrapper;

  it('should match the correct snapshot', () => {
    wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('Routes', () => {

    it('should render the Home container when at the root route', () => {
      wrapper = mount(
        <Provider store={store}>
          <MemoryRouter initialEntries={['/']}>
            <App />
          </MemoryRouter>
        </Provider>
      );
      expect(wrapper.find(Home)).toHaveLength(1);
    });

    it('should render the Share container when at route /project/:id', () => {
      wrapper = mount(
        <Provider store={store}>
          <MemoryRouter initialEntries={['/project/1']}>
            <App />
          </MemoryRouter>
        </Provider>
      );
      expect(wrapper.find(Share)).toHaveLength(1);
    });

    it('should render the Share container when at route /project/:id', () => {
      wrapper = mount(
        <Provider store={store}>
          <MemoryRouter initialEntries={['/palette/1']}>
            <App />
          </MemoryRouter>
        </Provider>
      );
      expect(wrapper.find(Share)).toHaveLength(1);
    });

    it('should render the NotFound container when at a route that is not recognized', () => {
      const wrapper = mount(
        <Provider store={store}>
          <MemoryRouter initialEntries={['/banana']}>
            <App />
          </MemoryRouter>
        </Provider>
      );
      expect(wrapper.find(NotFound)).toHaveLength(1);
    });

  });

});