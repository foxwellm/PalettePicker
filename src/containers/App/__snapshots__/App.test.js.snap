// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the correct snapshot 1`] = `
<div>
  <Connect(Header) />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      path="/project/:id"
      render={[Function]}
    />
    <Route
      path="/palette/:id"
      render={[Function]}
    />
    <Route
      component={[Function]}
    />
  </Switch>
</div>
`;
